### PR TITLE
Adjust bsend rate to match # of enabled streams

### PR DESCRIPTION
**Pull request by bmerry**
This updates spead2 to a newer version that implements the necessary feature.

Checklist (if not applicable, edit to add `(N/A)` and mark as done):

- [x] If dependencies are added/removed: update `setup.cfg` and `.pre-commit-config.yaml`
- [x] (N/A) If modules are added/removed: use `sphinx-apidoc -efo doc/ src/` to update files in `doc/`
- [x] Ensure copyright notices are present and up-to-date
- [x] (N/A) If qualification tests are changed: attach a sample qualification report
- [x] (N/A) If design has changed: ensure documentation is up to date
- [x] (N/A) If ICD-defined sensors have been added: update `fake_servers.py` in katsdpcontroller to match

Closes NGC-1143.

### CODE REVIEW

File: src/katgpucbf/xbgpu/bsend.py [hunk 1]
```diff
@@ -172,10 +172,18 @@ def send(
...
         """
-        if any(send_stream.tx_enabled):
+        n_enabled = sum(send_stream.tx_enabled)
+        rate = send_stream.bytes_per_second_per_beam * n_enabled
+        if n_enabled > 0:
             send_futures = []
             for frame in self._frames:
-                heaps_to_send = [heap for heap, enabled in zip(frame.heaps, send_stream.tx_enabled) if enabled]
+                # TODO: building this list every time may be too expensive.
+                # Consider caching it and invalidating when streams are enabled/disabled.
```
Comment (amishatishpatel) [inline src/katgpucbf/xbgpu/bsend.py:181]: I might have to request you create a ticket for this, please.

File: src/katgpucbf/xbgpu/bsend.py [hunk 2]
```diff
@@ -172,10 +172,18 @@ def send(
...
+                heaps_to_send = [
+                    spead2.send.HeapReference(heap, substream_index=i, rate=rate)
+                    for i, (heap, enabled) in enumerate(zip(frame.heaps, send_stream.tx_enabled))
+                    if enabled
+                ]
```
Comment (amishatishpatel) [inline src/katgpucbf/xbgpu/bsend.py:186]: I see, I see.

**Review by amishatishpatel — CHANGES_REQUESTED** [review]
Just one request to create a ticket to track that new TODO. Otherwise, looks good to me; very nifty.

**Comment by bmerry** [comment]
> I looked over at spead2 and see you're now at version 4.3.1 (as of this morning). Will the version here need to be bumped again (this PR or separately)?

I'm bumping it in a separate PR.

File: src/katgpucbf/xbgpu/bsend.py (hunk 1 above)
Comment (bmerry) [reply src/katgpucbf/xbgpu/bsend.py]: Done.

**Review by amishatishpatel — APPROVED** [review]

**Pull request merged**